src/NHSD.BuyingCatalogue.Identity.Api/Controllers/AccountController.cs

```diff
@@ -99,6 +105,28 @@ public async Task<IActionResult> Login(LoginViewModel viewModel)
             throw new Exception("Invalid return URL");
         }
 
+        [HttpGet]
+        public async Task<IActionResult> Logout(string logoutId)
+        {
+            return await Logout(new LogoutViewModel
+            {
+                LogoutId = logoutId
+            });
+        }
+
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public async Task<IActionResult> Logout(LogoutViewModel logoutViewModel)
+        {
+            logoutViewModel.ThrowIfNull(nameof(logoutViewModel));
+
+            LogoutRequest logoutRequest = await _logoutService.GetLogoutRequest(logoutViewModel.LogoutId);
+
+            await _logoutService.SignOutAsync(logoutRequest);
+
+            return Redirect(logoutRequest?.PostLogoutRedirectUri);
```

Comment:
As `Redirect` will throw if the URL passed to it is null or empty is it worth a change of approach here? The other option is to allow `Redirect` to throw and the user should be directed to the error page (once that change is merged in).

Reply:
I think the Redirect will throw an exception. I'm not sure what to do if the PostLogoutRedirectUri is null other than redirecting to the route '~/' but either way is not great for the user. I suppose the error page would give some feedback as least.

Reply:
Probably worth a follow-up task...